File: common/src/main/java/com/wdz/common/net/http/AddCookiesInterceptor.java
package com.wdz.common.net.http;

import android.util.Log;

import com.wdz.common.util.MmkvUtils;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class AddCookiesInterceptor implements Interceptor {
    private final String TAG = this.getClass().getSimpleName();
    @Override
    public Response intercept(Chain chain) throws IOException {
        Request.Builder builder = chain.request().newBuilder();
        Log.i(TAG, "intercept: "+MmkvUtils.getCookie());
        builder.addHeader("Cookie", MmkvUtils.getCookie());
        return chain.proceed(builder.build());
    }
}

File: kt_common/src/main/java/com/wdz/ktcommon/http/interceptor/AddCookiesInterceptor.kt
package com.wdz.ktcommon.http.interceptor

import android.util.Log
import com.wdz.ktcommon.utils.getCookie
import okhttp3.Interceptor
import okhttp3.Response

/**

 * @Author dezhi.wang

 * @Date 2021/3/31 10:21

 */
class AddCookiesInterceptor:Interceptor {
    private val TAG = this::class.simpleName
    override fun intercept(chain: Interceptor.Chain): Response {
        val builder = chain.request().newBuilder()
        Log.i(TAG, "intercept: " + getCookie())
        builder.addHeader("Cookie", getCookie())
        return chain.proceed(builder.build())
    }
}